javalib/src/main/scala/java/lang/IEEE754Helpers.scala

```diff
@@ -0,0 +1,68 @@
+package java.lang
+
+import scalanative.unsafe._
+import scalanative.libc.errno
+
+import scala.annotation.{switch, tailrec}
+import scalanative.posix.errno.ERANGE
+
+private[java] object IEEE754Helpers {
+
+  // Java parseDouble() and parseFloat() allow characters at & after
+  // after where C strtod() or strtof() stops. Continuous trailing whitespace
+  // with or without an initial 'D', 'd', 'F', 'f' size indicator
+  // is allowed.
+  // The whitespace can include tabs ('\t') and even newlines! ('\n').
+  // Perhaps even Unicode....
+  //
+  // The Java trim method is used to ensure that Java rules are followed.
+  // It might not be the most memory & runtime efficient, but if execution
+  // is in this routine, it is already on a slow path.
+
+  @tailrec
+  def vetIEEE754Tail(tail: String): Boolean = {
+
+    if (tail.length <= 0) {
+      true
+    } else {
+      val rest = tail.tail.trim
+      (tail(0): @scala.annotation.switch) match {
+        case 'D' | 'd' => vetIEEE754Tail(rest)
+        case 'F' | 'f' => vetIEEE754Tail(rest)
+        case _         => (tail.trim.length == 0)
+      }
+    }
+  }
```

Comment:
Since we know that chars in the range [0-32] are admissible, we can easily write this without `trim`, not causing any more allocation than necessary:
```scala
private def vetIEEE754Tail(tail: String): Boolean = {
  val len = tail.length
  if (len == 0) true
  else {
    var i = 0
    if ((tail.charAt(0) & 0xdd) == 0x44) // magic: is first char one of D d F f
      i = 1
    while (i != len) {
      if ((tail.charAt(i) & 0xff) > ' ')
        return false
      i += 1
    }
    true
  }
}
```
And to avoid even allocating the substring at call site, we can take the start index instead of assuming that we have to start at 0:
```scala
private def vetIEEE754Tail(tail: String, start: Int): Boolean = {
  val len = tail.length
  if (len == start) true
  else {
    var i = start
    if ((tail.charAt(i) & 0xdd) == 0x44) // magic: is first char one of D d F f
      i += 1
    while (i != len) {
      if ((tail.charAt(i) & 0xff) > ' ')
        return false
      i += 1
    }
    true
  }
}
```

Reply:
I extensively re-worked the implementation based on review comments, sample code, the review 
concern with minimizing allocations and a years worth of experience & understanding.

The idea was to do the math in the C and C byte realm rather than improperly conflating Scala/Java Strings.
I also added a few unit-test tests for corner cases which I found tricky or hard to remember.